File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    let answer = 1;
    let init = section[0]
    
    for(var i=0; i<section.length; i++) {
        if(init+(m-1) < section[i]){
            init = section[i]
            answer += 1
        }
    }
    return answer;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 1
    # section에서 첫번째 구간은 무조건 포함 시킨다. 
    init = section[0]
    
    for s in section:
        if init + (m-1) < s:
            init = s
            answer += 1
        
    return answer